src/components/Root/Root.tsx

```diff
@@ -1,52 +1,50 @@
 import React, { FC, useEffect, useState } from 'react';
-import { useTranslation } from 'react-i18next';
 
 import ErrorPage from '#components/ErrorPage/ErrorPage';
 import AccountModal from '#src/containers/AccountModal/AccountModal';
-import { IS_DEMO_MODE, IS_DEVELOPMENT_BUILD, IS_PREVIEW_MODE } from '#src/utils/common';
+import { IS_DEMO_MODE, IS_DEVELOPMENT_BUILD, IS_PREVIEW_MODE, IS_PROD_MODE } from '#src/utils/common';
 import DemoConfigDialog from '#components/DemoConfigDialog/DemoConfigDialog';
 import DevConfigSelector from '#components/DevConfigSelector/DevConfigSelector';
 import AppRoutes from '#src/containers/AppRoutes/AppRoutes';
 import registerCustomScreens from '#src/screenMapping';
 import LoadingOverlay from '#components/LoadingOverlay/LoadingOverlay';
-import { ConfigError, SettingsError } from '#src/utils/error';
 import { useBootstrapApp } from '#src/hooks/useBootstrapApp';
 
 const IS_DEMO_OR_PREVIEW = IS_DEMO_MODE || IS_PREVIEW_MODE;
 
-// This is moved to a separate, parallel component to reduce rerenders
-const RootLoader = ({ onReady }: { onReady: () => void }) => {
-  const { t } = useTranslation('error');
+const ProdContentLoader = ({ query }: { query: ReturnType<typeof useBootstrapApp> }) => {
+  const { isLoading, error } = query;
 
-  const { data, isLoading, error, isSuccess } = useBootstrapApp(onReady);
-  const { configSource } = data || {};
+  if (isLoading) {
+    return <LoadingOverlay />;
+  }
+
+  if (error) {
+    return <ErrorPage title={error?.payload?.title} message={error?.payload?.description} error={error} helpLink={error?.payload?.helpLink} />;
```

Comment:
I think there is no need to optional chain here. `Error` is of type `AppError | null` and all `payload` properties are non-optional.

```tsx
    return <ErrorPage title={error.payload.title} message={error.payload.description} error={error} helpLink={error.payload.helpLink} />;
```

Is it still needed to pass the `error` prop as well?

Reply:
Yeah, you are right, for production we don't really need error's object, but we can still pass it for dev/demo/prew.